sources/code/main/modules/keybind-socket.ts

```diff
@@ -0,0 +1,81 @@
+import net from "net";
+import fs from "fs";
+import keycode from "keycode";
+import { app, ipcMain } from "electron/main";
+
+interface KeybindState {
+  _state?: Record<string, {
+    id: string;
+    action: string;
+    shortcut: [0, number, 4][];
+    managed: boolean;
+    context: string;
+  }>;
+}
+
+export default function bindKeybindSocket(window: Electron.BrowserWindow) {
+  if (!process.env["XDG_RUNTIME_DIR"]) {
```

Comment:
This should fallback to `/run/user/${UID}/` AFAIK (I think `UID` can be accessed in `process`?).